library/std/src/sys/net/connection/socket/unix.rs

```diff
@@ -485,14 +485,18 @@ impl Socket {
 
     // bionic libc makes no use of this flag
     #[cfg(target_os = "linux")]
-    pub fn set_deferaccept(&self, accept: u32) -> io::Result<()> {
-        setsockopt(self, libc::IPPROTO_TCP, libc::TCP_DEFER_ACCEPT, accept as c_int)
+    pub fn set_deferaccept(&self, accept: Duration) -> io::Result<()> {
+        let val = accept.as_secs();
+        if val > i32::MAX.try_into().unwrap() {
+            return Err(io::const_error!(io::ErrorKind::InvalidInput, "awakening time overflow"));
+        }
```

Comment:
I meant a `try_into()` on the seconds value 😄.

```suggestion
        let Ok(val) = accept.as_secs().try_into() else {
            return Err(io::const_error!(io::ErrorKind::InvalidInput, "awakening time overflow"));
        };
```